c_glib/gandiva-glib/node.cpp

```diff
@@ -1387,6 +1393,156 @@ ggandiva_if_node_new(GGandivaNode *condition_node,
                                   return_type);
 }
 
+
+typedef struct GGandivaBooleanNodePrivate_ {
+  GList *parameters;
+} GGandivaBooleanNodePrivate;
+
+G_DEFINE_TYPE_WITH_PRIVATE(GGandivaBooleanNode,
+                           ggandiva_boolean_node,
+                           GGANDIVA_TYPE_NODE)
+
+#define GGANDIVA_BOOLEAN_NODE_GET_PRIVATE(object)      \
+  static_cast<GGandivaBooleanNodePrivate *>(           \
+    ggandiva_boolean_node_get_instance_private(        \
+      GGANDIVA_BOOLEAN_NODE(object)))                  \
+
+static void
+ggandiva_boolean_node_dispose(GObject *object)
+{
+  auto priv = GGANDIVA_BOOLEAN_NODE_GET_PRIVATE(object);
+
+  if (priv->parameters) {
+    for (auto node = priv->parameters; node; node = g_list_next(node)) {
+      auto parameter = GGANDIVA_NODE(node->data);
+      g_object_unref(parameter);
+    }
+    g_list_free(priv->parameters);
+    priv->parameters = nullptr;
+  }
+
+  G_OBJECT_CLASS(ggandiva_boolean_node_parent_class)->dispose(object);
+}
+
+static void
+ggandiva_boolean_node_init(GGandivaBooleanNode *boolean_node)
+{
+  auto priv = GGANDIVA_BOOLEAN_NODE_GET_PRIVATE(boolean_node);
+  priv->parameters = nullptr;
+}
+
+static void
+ggandiva_boolean_node_class_init(GGandivaBooleanNodeClass *klass)
+{
+  auto gobject_class = G_OBJECT_CLASS(klass);
+
+  gobject_class->dispose = ggandiva_boolean_node_dispose;
+}
+
+
+G_DEFINE_TYPE(GGandivaAndNode,
+              ggandiva_and_node,
+              GGANDIVA_TYPE_BOOLEAN_NODE)
+
+static void
+ggandiva_and_node_init(GGandivaAndNode *and_node)
+{
+}
+
+static void
+ggandiva_and_node_class_init(GGandivaAndNodeClass *klass)
+{
+}
+
+/**
+ * ggandiva_and_node_new:
+ * @parameters: (element-type GGandivaNode): The parameters of the AND node.
+ *
+ * Returns: A newly created #GGandivaAndNode for the AND expression.
+ *
+ * Since: 1.0.0
+ */
+GGandivaAndNode *
+ggandiva_and_node_new(GList *parameters)
+{
+  std::vector<std::shared_ptr<gandiva::Node>> gandiva_nodes;
+  for (auto node = parameters; node; node = g_list_next(node)) {
+    auto gandiva_node = ggandiva_node_get_raw(GGANDIVA_NODE(node->data));
+    gandiva_nodes.push_back(gandiva_node);
+  }
+  auto gandiva_node = gandiva::TreeExprBuilder::MakeAnd(gandiva_nodes);
+  return ggandiva_and_node_new_raw(&gandiva_node,
+                                   parameters);
+}
+
+/**
+ * ggandiva_and_node_get_parameters:
```

Comment:
`parameters` is natural for function node but `parameters` for boolean node is strange for me.

How about using `children` like Gandiva does?

And how about defining `ggandiva_boolean_node_get_children()` instead of `ggandiva_{and,or}_node_get_children()`?

Reply:
I've fixed them.